process/process.go

```diff
@@ -202,32 +204,37 @@ func (p *Process) Run(ctx context.Context) error {
 		go func() {
 			p.logger.Debug("[Process] Starting to copy PTY to the buffer")
 
-			// Copy the pty to our writer. This will block until it
-			// EOF's or something breaks.
+			// Copy the pty to our writer. This will block until it EOFs or something breaks.
 			_, err = io.Copy(p.conf.Stdout, pty)
-			if e, ok := err.(*os.PathError); ok && e.Err == syscall.EIO {
-				// We can safely ignore this error, because
-				// it's just the PTY telling us that it closed
-				// successfully.  See:
-				// https://github.com/buildkite/agent/pull/34#issuecomment-46080419
-				err = nil
-			}
+			switch {
+			case errors.Is(err, syscall.EIO):
```

Comment:
I guess technically this catches error that are not `os.PathError` that also happen to wrap `syscall.EIO`. I think it's unlikely that `io.Copy` will create such a thing, but I'm happy to do a more specific unwrap if we think otherwise.